src/test_utils/client.rs

```diff
@@ -0,0 +1,111 @@
+// Copyright 2015 MaidSafe.net limited.
+//
+//
+// This SAFE Network Software is licensed to you under (1) the MaidSafe.net Commercial License,
+// version 1.0 or later, or (2) The General Public License (GPL), version 3, depending on which
+// licence you accepted on initial access to the Software (the "Licences").
+//
+// By contributing code to the SAFE Network Software, or to this project generally, you agree to be
+// bound by the terms of the MaidSafe Contributor Agreement, version 1.0.  This, along with the
+// Licenses can be found in the root directory of this project at LICENSE, COPYING and CONTRIBUTOR.
+//
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.
+//
+// Please review the Licences for the specific language governing permissions and limitations
+// relating to use of the SAFE Network Software.
+
+extern crate time;
+
+pub struct Client {
+    routing_client: ::routing_client::RoutingClient,
+    receiver: ::std::sync::mpsc::Receiver<::event::Event>,
+    public_id: ::public_id::PublicId,
+}
+
+impl Client {
+    pub fn new() -> Client {
+        let (sender, receiver) = ::std::sync::mpsc::channel::<::event::Event>();
+
+        let id = ::id::Id::new();
+        let public_id = ::public_id::PublicId::new(&id);
+        let routing_client = ::routing_client::RoutingClient::new(sender, Some(id));
+
+        debug!("Client name {:?}", public_id.clone());
+
+        Client {
+            routing_client: routing_client,
+            receiver: receiver,
+            public_id: public_id,
```

Comment:
perhaps consider storing/moving the Id into Client, so that we can rejoin the network later with the same Id